mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -559,6 +561,14 @@ abstract class PcCollector[T](
         Span(span.start, span.start + realName.length, point)
       else Span(point, span.end, point)
     else span
+
+  // We don't want to collect synthethic `map`, `withFilter` and `flatMap` in for-comprenhensions
+  private def isForComprehensionMethod(sel: Select): Boolean =
+    val name = sel.name.decoded
+    val syntheticName =
+      name == "map" || name == "withFilter" || name == "flatMap"
```

Comment:
Is this check necessary? Or can `wrongSpan` be true for e.g. `apply`?

Reply:
Also you should add "foreach".

Reply:
> Is this check necessary? Or can wrongSpan be true for e.g. apply?

Exactly , eg. in `<<ListB>>uffer(123)`, `<<>>` mark span of `apply`

Actually, I just noticed there is a bug in Scala 2:
```scala
val x = <<ListB>>uffer(1)
val y = ListBuffer.<<ap@@ply>>(2)
```
I will create an issue and look at it in a follow up